### PR TITLE
feat(nft-erc20-payments): ERC20 Payments working

### PR DESCRIPTION
**Pull request by jmichel-bc**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts [hunk 1]
```diff
@@ -176,6 +176,31 @@ export function getOrderHash(order: UnhashedOrder) {
...
+        throw new Error('Gas estimation failing consistently.')
+      }
+    } else {
+      console.log(JSON.stringify(e, null, 4))
+      console.log(error.code)
+    }
+    estimatedValue = txData.gasLimit
+  }
+  return estimatedValue
+}
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts:202]: Nice

File: packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts [hunk 2]
```diff
@@ -1150,14 +1174,15 @@ async function _initializeProxy(signer): Promise<string> {
...
-  const gasPrice = await signer.getGasPrice()
   const txnData = {
     from: accountAddress,
-    gasLimit: 410_000,
-    gasPrice: parseInt(gasPrice._hex)
+    gasLimit: 410_000
   }
-  const gasLimitEstimated = await wyvernProxyRegistry.estimateGas.registerProxy(txnData)
-  txnData.gasLimit = parseInt(gasLimitEstimated._hex)
+  txnData.gasLimit = await safeGasEstimation(
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts:1181]: should wrap calls to `safeGasEstimation` in a try catch since it can throw?

**Review by plondon — APPROVED** [review]

File: packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts (hunk 2 above)
Comment (Donone19) [reply packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts:1181]: What I need to do to spend it

File: packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts (hunk 1 above)
Comment (Donone19) [reply packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts:202]: Ok nice

**Pull request merged**